src/psyclone/domain/gocean/kernel/psyir.py

```diff
@@ -424,16 +431,24 @@ def _validate_iterates_over(value):
     @property
     def name(self):
         '''
-        :returns: the name of the symbol to use when lowering.
+        :returns: the name of the symbol that will contain the \
+            metadata when lowering.
         :rtype: str
+
         '''
         return self._name
 
     @name.setter
     def name(self, value):
         '''
-        :param str value: set the name of the symbol to use when lowering.
+        :param str value: set the name of the symbol that will contain \
+            the metadata when lowering.
+
```

Comment:
Please document the :raises:. Also, should we perhaps do the `re.compile` when the class is constructed?

Reply:
Oops and OK